src/udisksdaemon.c

```diff
@@ -358,12 +358,10 @@ udisks_daemon_constructed (GObject *object)
         }
     else
       {
-        /* a missing plugin is okay, calling functions from it will fail, but
-           until that happens, life will just be great */
-        for (plugin_p=plugins; *plugin_p; plugin_p++)
+        for (plugin_p = plugins; *plugin_p; plugin_p++)
```

Comment:
People are not going to like this. We already had some RFEs asking for libblockdev plugins to be optional in the past and this will break some of the niche use cases out there.

If you think this is the way to go, I won't stop you, but be prepared for some angry issues in the future :-)

Reply:
> I would be more worried about the hard `libnvme` requirement, but at this point distributions all picked up the new `nvme-cli-2.0` and `libnvme-1.0` pretty nicely.

It wouldn't be a a hard dependency if the plugin had been optional. But I remember people disliking dependency on _libkmod_ so _libnvme_ could be even worse.

Reply:
I mean, people would scream at every new dependency. The mission here is clear: ATA, SCSI and NVMe are considered a core functionality, it's pointless to make such things optional.